nodes/filter/nodes.go

```diff
@@ -0,0 +1,225 @@
+
+	"github.com/NubeDev/flow-eng/helpers/float"
+	"github.com/NubeDev/flow-eng/node"
+)
+
+type PreventNull struct {
+	*node.Spec
+	lastValue float64
+}
+
+type PreventEqualFloat struct {
+	*node.Spec
+	lastValue float64
+}
+
+type PreventEqualString struct {
+	*node.Spec
+	lastValue string
+}
+
+type OnlyBetween struct {
+	*node.Spec
+	lastValue float64
+}
+
+type OnlyGreater struct {
+	*node.Spec
+	lastValue float64
+}
+
+type OnlyLower struct {
+	*node.Spec
+	lastValue float64
+}
+
+type PreventDuplicates struct {
+	*node.Spec
+	lastValue float64
+}
+
+func NewPreventNull(body *node.Spec) (node.Node, error) {
+	body = node.Defaults(body, preventNull, category)
+	inputs := node.BuildInputs(node.BuildInput(node.In, node.TypeFloat, nil, body.Inputs))
+	outputs := node.BuildOutputs(node.BuildOutput(node.Out, node.TypeFloat, nil, body.Outputs))
+	body = node.BuildNode(body, inputs, outputs, nil)
+	body.SetHelp(fmt.Sprintln("This node filters ‘input’ values. All ‘input’ values are passed to ‘output’ EXCEPT ‘null’. Output the last known value if the input became null. lastValue is defaulted to be 0."))
+	return &PreventNull{body, 0}, nil
+}
+
+func NewPreventEqualFloat(body *node.Spec) (node.Node, error) {
+	body = node.Defaults(body, preventEqualFloat, category)
+	inputs := node.BuildInputs(node.BuildInput(node.In, node.TypeFloat, nil, body.Inputs), node.BuildInput(node.InputMatch, node.TypeFloat, nil, body.Inputs))
+	outputs := node.BuildOutputs(node.BuildOutput(node.Out, node.TypeFloat, nil, body.Outputs))
+	body = node.BuildNode(body, inputs, outputs, nil)
+	body.SetHelp(fmt.Sprintln("This node filters ‘input’ values. All ‘input’ values are passed to ‘output’ EXCEPT ‘input’ values which are equal to ‘match’. Output the last known input float value if two inputs equal to each other. Out put nil if any of the inputs are nil."))
+	return &PreventEqualFloat{body, 0}, nil
+}
+
+func NewPreventEqualString(body *node.Spec) (node.Node, error) {
+	body = node.Defaults(body, preventEqualString, category)
+	inputs := node.BuildInputs(node.BuildInput(node.In, node.TypeString, nil, body.Inputs), node.BuildInput(node.InputMatch, node.TypeString, nil, body.Inputs))
+	outputs := node.BuildOutputs(node.BuildOutput(node.Out, node.TypeString, nil, body.Outputs))
+	body = node.BuildNode(body, inputs, outputs, nil)
+	body.SetHelp(fmt.Sprintln("This node filters ‘input’ values. All ‘input’ values are passed to ‘output’ EXCEPT ‘input’ values which are equal to ‘match’. Output the last known input string value if two inputs equal to each other. Out put nil if any of the inputs are nil."))
+	return &PreventEqualString{body, "defaultString"}, nil
+}
+
+func NewOnlyBetween(body *node.Spec) (node.Node, error) {
+	body = node.Defaults(body, onlyBetween, category)
+	inputs := node.BuildInputs(node.BuildInput(node.In, node.TypeFloat, nil, body.Inputs), node.BuildInput(node.InMin, node.TypeFloat, nil, body.Inputs), node.BuildInput(node.InMax, node.TypeFloat, nil, body.Inputs))
```

Comment:
correct input names to match rubix-wires documentation. Leave output names as `node.Out`.